### PR TITLE
Improve Ruby SDK setup and update docs

### PR DESCRIPTION
**Pull request by ccschmitz**
## Summary

Updates the Ruby SDK and docs.

* Eliminates the need to set up the `around_action` to instrument requests by adding a `Railtie` and setting up the `around_action` as part of the initialization logic.
* Adds an `init` alias so you can call `Highlight.init` rather than `Highlight::H.new`.
* Fixes an issue where we were seeing warnings about incompatibility with `ActiveSupport::LoggerSilence`.
* Updates the Ruby/Rails documentation.
* Adds a `/for` page for Rails.
* Fixes some broken links on existing `/for` pages.

Closes HIG-4900

## How did you test this change?

Click tested locally in the E2E app + a local instance of the docs site.

## Are there any deployment considerations?

New SDK version going out. Will see if I can get recent articles published about the Ruby SDK updated.

## Does this work require review from our design team?

N/A

### CODE REVIEW

File: docs-content/getting-started/5_backend-logging/04_ruby/1_overview.md [hunk 1]
```diff
@@ -17,7 +17,4 @@ If you don't see one of your languages / frameworks below, reach out to us in ou
     <DocsCard title="Other Ruby Frameworks" href="./other.md">
         {"Integrate logging in other Ruby Frameworks."}
     </DocsCard>
-    <DocsCard title="Ruby OpenTelemetry" href="../../7_native-opentelemetry/3_logging.md">
-        {"Integrate logging with Native OpenTelemetry."}
-    </DocsCard>
```
Comment (ccschmitz) [inline docs-content/getting-started/5_backend-logging/04_ruby/1_overview.md:22]: @Vadman97 these docs linked to a page that showed JS example code. Was thinking it's better to hide these until we have some Ruby code examples for this. Is that cool with you?

Comment (Vadman97) [reply docs-content/getting-started/5_backend-logging/04_ruby/1_overview.md:22]: yeah, sounds great to me

File: sdk/highlight-ruby/highlight/lib/highlight.rb [hunk 2]
```diff
@@ -195,6 +199,8 @@ def trace_id_from_headers(headers)
   end
 
   class Logger < ::Logger
+    include ActiveSupport::LoggerSilence if defined?(::ActiveSupport::LoggerSilence)
```
Comment (Vadman97) [inline sdk/highlight-ruby/highlight/lib/highlight.rb:202]: thanks for the fix!

File: sdk/highlight-ruby/highlight/lib/highlight.rb [hunk 3]
```diff
@@ -18,6 +18,10 @@ def on_start(span, parent_context)
     end
   end
 
+  def self.init(project_id, environment: '', otlp_endpoint: OTLP_HTTP, &block)
```
Comment (Vadman97) [inline sdk/highlight-ruby/highlight/lib/highlight.rb:21]: nice, more in line with the other sdks

**Review by Vadman97 — APPROVED** [review]

**Pull request merged**